ref_builder/cli/otu.py

```diff
@@ -341,3 +349,24 @@ def plan_rename_segment(
         )
     except ValueError as e:
         click.echo(e, err=True)
+
+
+def _get_otu_id_from_other_identifier(repo: Repo, identifier: str) -> UUID | None:
+    """Return an OTU id from the repo if identifier matches a single OTU.
+    Return None if no matching OTU is found, raise a ValueError if >1 OTU is found.
+    """
+    try:
+        identifier = int(identifier)
+        return repo.get_otu_id_by_taxid(identifier)
+    except ValueError:
+        pass
```

Comment:
This avoids breaking the typing of  `identifier`.
```suggestion
    try:
        return repo.get_otu_id_by_taxid(int(identifier))
    except ValueError:
        pass
```
I'd also check that the `ValueError` is related to the `int` type conversion. Alternatively:

```py
taxid = None

with contextlib.suppress(ValueError):
    taxid = int(identifier)

if taxid:
    return repo.get_otu_id_by_taxid(identifier)
```